runtime/lua/vim/lsp.lua

```diff
@@ -115,6 +115,13 @@ local format_line_ending = {
   ["mac"] = '\r',
 }
 
+---@private
+---@param bufnr (number)
+---@returns (string)
+local function buf_get_line_ending(bufnr)
+  return format_line_ending[nvim_buf_get_option(bufnr, 'fileformat')] or '\n'
```

Comment:
in what scenario do we need the fallback?

Reply:
It is simply written defensively, AFAIK it is impossible to set fileformat to an invalid value